File: BusinessLogic/PayIn.Application.Dto/Arguments/ControlIncident/ControlIncidentCreateManualCheckArguments.cs
using PayIn.Common;
using System;
using Xp.Common.Dto.Arguments;

namespace PayIn.Application.Dto.Arguments.ControlIncident
{
	public class ControlIncidentCreateManualCheckArguments : IArgumentsBase
	{
		public IncidentType IncidentType { get; set; }
		public String       Observations { get; set; }
		public ControlIncidentCreateManualCheckArguments_IncidentItem Item { get; set; }

		#region Constructors
		public ControlIncidentCreateManualCheckArguments(IncidentType incidentType, String observations, ControlIncidentCreateManualCheckArguments_IncidentItem item)
		{
			IncidentType = incidentType;
			Observations = observations ?? "";
			Item = item;
		}
		#endregion Constructors
	}
}

File: AndroidApp/PayIn/app/src/main/java/com/mobile/application/payin/dto/arguments/ControlIncidentCreateManualCheckArguments.java
package com.mobile.application.payin.dto.arguments;

import java.io.Serializable;
import java.util.ArrayList;

public class ControlIncidentCreateManualCheckArguments implements Serializable {
    public int          IncidentType;
    public String       Observations;
    public IncidentItem Item;

    public ControlIncidentCreateManualCheckArguments() { }

    public static class IncidentItem implements Serializable {
        public String            Date;
        public String            Image;
        public Double            Latitude;
        public Double            Longitude;
        public int               Id;
        public Integer           CheckPointId;
        public Integer           CheckId;
        public ArrayList<Assign> Assigns;

        public IncidentItem() {
            Assigns = new ArrayList<>();
        }
    }

    public static class Assign implements Serializable {
        public int Id;
        public ArrayList<Value> Values;

        public Assign() {
            Values = new ArrayList<>();
        }
    }

    public static class Value implements Serializable {
        public int     Id;
        public String  ValueString;
        public Double  ValueNumeric;
        public Boolean ValueBool;
        public String  ValueDateTime;
        public String  ValueImage;

        public Value() { }
    }
}
